### PR TITLE
CFE-4310: Fixed static-check job by forcing inclusion of config.h

### PR DESCRIPTION
**Pull request by craigcomstock**
Apparently config.h is not included from platform.h because HAVE_CONFIG_H is not defined or not checked as a possibility soon enough by cppcheck.

Ticket: CFE-4310
Changelog: none

### CODE REVIEW

**Comment by craigcomstock** [comment]
when running locally cppcheck found a few problems, I will try and fix. I suspect this static_check has been failing for some time?

```
cf-agent/verify_users_pam.c:616:5: error: Used file that is not opened. [useClosedFile]
    fclose(edit_fd);
    ^
cf-agent/verify_users_pam.c:619:5: error: Used file that is not opened. [useClosedFile]
    fclose(passwd_fd);
    ^
cf-check/diagnose.c:285:5: error: Resource leak: f_result [resourceLeak]
    }
    ^
```

**Comment by craigcomstock** [comment]
good, previous ci run before I pushed some C code fixes ran now and showed the errors I saw locally :) https://github.com/cfengine/core/actions/runs/7507854424/job/20442321164

File: cf-agent/verify_users_pam.c [hunk 1]
```diff
@@ -593,7 +593,9 @@ static bool ChangePasswordHashUsingLckpwdf(const char *puser, const char *passwo
     }
 
     fclose(edit_fd);
+    edit_fd = NULL;
```
Comment (craigcomstock) [inline cf-agent/verify_users_pam.c:596]: could add a comment here saying "so we know to not close it later"

File: cf-agent/verify_users_pam.c [hunk 2]
```diff
@@ -613,10 +615,10 @@ static bool ChangePasswordHashUsingLckpwdf(const char *puser, const char *passwo
     goto unlock_passwd;
 
 close_both:
-    fclose(edit_fd);
+    if (edit_fd != NULL) fclose(edit_fd);
```
Comment (larsewi) [inline cf-agent/verify_users_pam.c:618]: CONTRIBUTING.md requires us to always use curly braces in if statements

**Review by larsewi — APPROVED** [review]

File: cf-agent/verify_users_pam.c (hunk 2 above)
Comment (craigcomstock) [reply cf-agent/verify_users_pam.c:618]: yeah, I felt trepidation writing that, should have trusted my gut ;) willfix.

**Review by larsewi — APPROVED** [review]

**Comment by craigcomstock** [comment]
cherry picks
https://github.com/cfengine/core/pull/5431
https://github.com/cfengine/core/pull/5430

**Pull request merged**